File: src/main/java/com/github/doomsdayrs/jikan4java/types/support/basic/BasicObject.java
package com.github.doomsdayrs.jikan4java.types.support.basic;

/*
 * This file is part of Jikan4java.
 *
 * Jikan4java is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jikan4java is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
 */


import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.github.doomsdayrs.jikan4java.core.Retriever;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import org.json.simple.parser.JSONParser;

/**
 * com.github.doomsdayrs.jikan4java.types.support.basic
 * 17 / November / 2019
 *
 * @author github.com/doomsdayrs
 */
public class BasicObject extends Retriever {
    @JsonProperty("mal_id")
    public int mal_id;
    @JsonProperty("url")
    public String url;
    @JsonProperty("image_url")
    public String image_url;
    @JsonProperty("name")
    public String name;

    public BasicObject(int mal_id, String url, String image_url, String name) {
        this.mal_id = mal_id;
        this.url = url;
        this.image_url = image_url;
        this.name = name;
    }

    @Override
    public String toString() {
        return "BasicObject{" +
                "mal_id=" + mal_id +
                ", url='" + url + '\'' +
                ", image_url='" + image_url + '\'' +
                ", name='" + name + '\'' +
                '}';
    }
}

File: src/main/java/com/github/doomsdayrs/jikan4java/model/support/basic/BasicObject.kt
package com.github.doomsdayrs.jikan4java.model.support.basic

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.doomsdayrs.jikan4java.core.Retriever

/*
 * This file is part of Jikan4java.
 *
 * Jikan4java is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jikan4java is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
 */ /**
 * com.github.doomsdayrs.jikan4java.types.support.basic
 * 17 / November / 2019
 *
 * @author github.com/doomsdayrs
 */
open class BasicObject(
		@field:JsonProperty("mal_id") val mal_id: Int,
		@field:JsonProperty("url") val url: String?,
		@field:JsonProperty("image_url") val image_url: String?,
		@field:JsonProperty("name") val name: String?
) : Retriever()